grabbit/src/main/groovy/com/twcable/grabbit/client/batch/ClientBatchJobContext.groovy

```diff
@@ -25,17 +27,96 @@ import javax.jcr.Session
  * and {@link Session} in ThreadLocal.
  */
 @CompileStatic
+@Slf4j
 class ClientBatchJobContext {
 
-    static final ThreadLocal<ClientBatchJobContext> THREAD_LOCAL = new ThreadLocal<ClientBatchJobContext>()
+    private static final ThreadLocal<ClientBatchJobContext> _THREAD_LOCAL = new ThreadLocal<ClientBatchJobContext>()
 
-    final InputStream inputStream
-    final Session session
+    private final InputStream inputStream
+    private final Session session
 
-
-    ClientBatchJobContext(InputStream inputStream, Session session) {
+    private ClientBatchJobContext(InputStream inputStream, Session session) {
         this.inputStream = inputStream
         this.session = session
     }
 
+    /**
+     * @return the job's JCR session, or null if it has not been started/set
+     */
+    static Session getSession() {
+        _THREAD_LOCAL.get()?._getSession()
```

Comment:
It returns the ClientBatchJobContext set on the thread, or initialValue(); which since we don't override, the parent returns null.